include/kokkos-utils/impl/type_traits.hpp

```diff
@@ -32,6 +32,10 @@ template <typename T, template <typename...> class U>
 concept InstanceOf = is_instance_of_v<T, U>;
 ///@}
 
+//! Check if @p T is the same as one of the types @p Ts.
+template <typename T, typename... Ts>
+concept IsTypeOneOf = ((std::same_as<T, Ts>) || ...);
```

Comment:
Not tested but worth a try:
```suggestion
template <typename T, typename... Ts>
concept IsTypeOneOf = std::disjunction<(std::same_as<T, Ts>, ...)>;
```

According to https://en.cppreference.com/w/cpp/types/disjunction:
```
The short-circuit instantiation differentiates disjunction from fold expressions: a fold expression like (... || Bs::value) instantiates every B in Bs, while std::disjunction_v<Bs...> stops instantiation once the value can be determined. This is particularly useful if the later type is expensive to instantiate or can cause a hard error when instantiated with the wrong type. 
```

Reply:
OK! That's a nice suggestion!